src/exchanges/adapters/ZeroExV3Adapter.sol

```diff
@@ -0,0 +1,256 @@
+
+        // Execute take order on exchange
+        IZeroExV3(_targetExchange).fillOrder(_order, _fillExpectedAmounts[1], _signature);
+    }
+
+    function __formatFillTakeOrderArgs(
+        address _targetExchange,
+        address[8] memory _orderAddresses,
+        uint256[8] memory _orderValues
+    )
+        internal
+        view
+        returns (address[] memory, uint256[] memory)
+    {
+        address[] memory fillAssets = new address[](4);
+        fillAssets[0] = _orderAddresses[2]; // maker asset
+        fillAssets[1] = _orderAddresses[3]; // taker asset
+        fillAssets[2] = __getNativeAssetAddress(); // protocol fee
+        fillAssets[3] = _orderAddresses[7]; // taker fee asset
+
+        uint256[] memory fillExpectedAmounts = new uint256[](4);
+        fillExpectedAmounts[0] = __calculateExpectedFillAmount(
+            _orderValues[1],
+            _orderValues[0],
+            _orderValues[6]
+        ); // maker fill amount; calculated relative to taker fill amount
+        fillExpectedAmounts[1] = _orderValues[6]; // taker fill amount
+        fillExpectedAmounts[2] = __calcProtocolFeeAmount(_targetExchange); // protocol fee
+        fillExpectedAmounts[3] = __calculateExpectedFillAmount(
+            _orderValues[1],
+            _orderValues[3],
+            _orderValues[6]
+        ); // taker fee amount; calculated relative to taker fill amount
+
+        return (fillAssets, fillExpectedAmounts);
+    }
+
+    function __getAssetProxy(address _targetExchange, bytes memory _assetData)
+        internal
+        view
+        returns (address assetProxy_)
+    {
+        bytes4 assetProxyId;
+        assembly {
+            assetProxyId := and(mload(
+                add(_assetData, 32)),
+                0xFFFFFFFF00000000000000000000000000000000000000000000000000000000
+            )
+        }
+        assetProxy_ = IZeroExV3(_targetExchange).getAssetProxy(assetProxyId);
+    }
+
+    function __getAssetAddress(bytes memory _assetData)
+        internal
+        pure
+        returns (address assetAddress_)
+    {
+        assembly {
+            assetAddress_ := mload(add(_assetData, 36))
+        }
+    }
+
+    function __validateTakeOrderParams(
+        address _targetExchange,
+        address[8] memory _orderAddresses,
+        uint256[8] memory _orderValues,
+        bytes[4] memory _orderData,
+        bytes memory _signature
+    )
+        internal
+        view
+    {
+        require(
+            __getAssetAddress(_orderData[0]) == _orderAddresses[2],
+            "__validateTakeOrderParams: makerAssetData does not match address"
+        );
+        require(
+            __getAssetAddress(_orderData[1]) == _orderAddresses[3],
+            "__validateTakeOrderParams: takerAssetData does not match address"
+        );
+        require(
+            _orderValues[6] <= _orderValues[1],
+            "__validateTakeOrderParams: taker fill amount greater than max order quantity"
```

Comment:
is this comment accurate?

Reply:
I believe so? 'require that the taker fill amount is less than or equal to the order's full taker amount'. That latter part if what we're referring to as "max order quantity". Is the issue that term?

Reply:
Comment is fine, my bad